editor/src/clj/editor/material.clj

```diff
@@ -228,9 +226,12 @@
        :columns (let [semantic-type-values (protobuf/enum-values Graphics$VertexAttribute$SemanticType)
                       data-type-values (protobuf/enum-values Graphics$VertexAttribute$DataType)
                       coordinate-space-values (protobuf/enum-values Graphics$CoordinateSpace)
+                      vector-type-values (protobuf/enum-values Graphics$VertexAttribute$VectorType)
+                      vertex-step-function (protobuf/enum-values Graphics$VertexStepFunction)
                       default-semantic-type :semantic-type-none
-                      default-element-count 3
-                      default-values (graphics/resize-doubles (vector-of :double) default-semantic-type default-element-count)]
+                      default-vector-type :vector-type-vec4
+                      default-values (graphics/resize-doubles (vector-of :double) default-semantic-type default-vector-type)
```

Comment:
What we currently have is exactly how editing matrix uniforms work afaik. If you add a "user matrix" in the constants table it will default to zeroes and only show the top row of the matrix. That top row is editable so I think, *for now*, we should have the same behaviour and really really make sure to fix this soon :)